### DISCUSSION TITLE
Clocked partition initialization based on model input signals

### DISCUSSION THREAD

**Discussion by MatteoSchiavon** [OpenModelica/OpenModelica#11704] [Q&A] [answered]
I have a **PLL (Phase Locked Loop) component model** which has 2 clocked input variables (ualpha, ubeta) and 1 clocked output variable (theta).
One of the components of the **PLL model** is of block class "Integrator" and an equation in the **PLL model** defines "connect(integrator.y, theta)", which means that the output of the integrator component will be the output of the **PLL**.

_"block Integrator
  extends Modelica.Clocked.RealSignals.Interfaces.PartialClockedSISO;
  Real x "Discrete integrator state";
equation
   x = previous(x) + u*interval(u);
   y = x;"_

To have a correct initialization of the model (PLL locked), the output angle theta must be initialized according to the value of the inputs "ualpha" and "ubeta". According to Modelica Language Specification 16.9, I am trying to initialize the model in the following way: 
_Integrator integrator(x.start=atan2(ubeta,ualpha))_ 
and obviously, I get the following error: 
"[Component start of variability parameter has binding 'atan2(ubeta, ualpha)' of higher variability continuous.]"
Since we cannot use the initial eqution section for clocked partitions, is there a way to solve this initialization problem?

**Comment by casella** [comment, marked as answer]
You can always define parameters with `fixed = false` attribute and then assign them to continuous-time variables in the initial equation section, e.g.:
```modelica
model M
  Real x = sin(time);
  parameter Real x_start(fixed = false);
initial equation
  x_start = x;
end M;
```

**Reply by MatteoSchiavon** [reply]
Thanks a lot, that solves the issue :) So in case my clocked state variable is very "low level" in the components hierarchy of the model, I'll have to define those not-fixed parameters at high level in the model where continuous time variables are available, and propagate them down to the clocked components?

**Reply by casella** [reply]
Not necessarily. The boundaries of clocked partitions are dictated by the equations that relate their variables, so you could also in principle have continuous variables in a block that mostly contains clocked variables. But I guess it makes sense to compute these initial value parameters where the continuous variables are found, and then just propagate the parameters.

**Reply by MatteoSchiavon** [reply]
Got it! Thanks a lot :)
